plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/condition/TableHasRows.java

```diff
@@ -7,33 +7,91 @@
  * 
  * Contributors: 
  * Red Hat, Inc. - initial API and implementation 
- ******************************************************************************/ 
+ ******************************************************************************/
 package org.jboss.reddeer.swt.condition;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.hamcrest.Matcher;
 import org.jboss.reddeer.common.condition.AbstractWaitCondition;
 import org.jboss.reddeer.swt.api.Table;
+import org.jboss.reddeer.swt.api.TableItem;
 
 /**
  * Condition is met when table has rows.
  * 
- * @author Rastislav Wagner
+ * @author Rastislav Wagner, Josef Kopriva
  */
 public class TableHasRows extends AbstractWaitCondition {
 	private final Table table;
+	private final Matcher<?> matcher;
+	private final Integer column;
 
 	/**
-	 * Constructs TableHasRows wait condition.
-	 * Condition is met when table has any rows.
+	 * Constructs TableHasRows wait condition. Condition is met when table has
+	 * any rows.
 	 * 
-	 * @param table table which should contain rows to the condition be met
+	 * @param table
+	 *            table which should contain rows to the condition be met
 	 */
 	public TableHasRows(Table table) {
+		this(table, null, null);
+	}
+
+	/**
+	 * Constructs TableHasRow wait condition. Condition is met when table has
+	 * specific row matched with matcher.
+	 * 
+	 * @param table
+	 *            table which should contain row to the condition be met
+	 * @param matchers
+	 *            matchers - e.g. Regexmatcher, custom matcher...
+	 */
+	public TableHasRows(Table table, Matcher<?> matcher) {
+		this(table, null, matcher);
+	}
+
+	/**
+	 * Constructs TableHasRow wait condition. Condition is met when table has
+	 * specific row and value in specific column matched with matcher.
+	 * 
+	 * @param table
+	 *            table which should contain row to the condition be met
+	 * @param column
+	 *            column in which should be used matchers
+	 * @param matchers
+	 *            matchers - e.g. Regexmatcher, custom matcher...
+	 */
+	public TableHasRows(Table table, Integer column, Matcher<?> matcher) {
 		this.table = table;
+		this.matcher = matcher != null ? matcher : null;
+		this.column = column != null ? column.intValue() : null;
+	}
+
+	private List<TableItem> search(int begin, int end) {
+		List<TableItem> matchedItems = new ArrayList<TableItem>();
+
+		for (TableItem item : table.getItems()) {
+			for (int i = begin; i < end; i++) {
+				if (matcher.matches(item.getText(i))) {
```

Comment:
In general, this will be a better solution. But in this case it is not necessary. Just out of curiosity, is there any item matcher without matching a text of the the item?